### ISSUE TITLE
Maximum call stack size exceeded (package: hist)

### ISSUE THREAD

**Issue by bfis** [microsoft/pylance-release#1671]
## Environment data

-   Language Server version: 2021.8.1 (pyright abd41b72)
-   OS and version: Ubuntu 20.10
-   Python version: 3.8.10

## Issue

Pylance does not work due to an `RangeError: Maximum call stack size exceeded` when using the [`hist`-package](https://github.com/scikit-hep/hist).

## Logs

<details>
  <summary>Pylance trace log</summary>

```
[Info  - 4:11:48 PM] Pylance language server 2021.8.1 (pyright abd41b72) starting
[Info  - 4:11:48 PM] Server root directory: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist
[Info  - 4:11:48 PM] No configuration file found.
[Info  - 4:11:48 PM] pyproject.toml file found at /home/USER/repos/REPO.
[Info  - 4:11:48 PM] Setting pythonPath for service "REPO": "/home/USER/.local/bin/python"
[Info  - 4:11:48 PM] Loading pyproject.toml file at /home/USER/repos/REPO/pyproject.toml
[Error - 4:11:48 PM] Pyproject file "/home/USER/repos/REPO/pyproject.toml" is missing "[tool.pyright] section.
[Warn  - 4:11:48 PM] stubPath /home/USER/repos/REPO/typings is not a valid directory.
[Info  - 4:11:48 PM] Assuming Python version 3.8
[Info  - 4:11:48 PM] Assuming Python platform Linux
Search paths for /home/USER/repos/REPO
  /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib
  /home/USER/repos/REPO
  /home/USER/repos/REPO/typings
  /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stubs/...
  /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/bundled/stubs
  /usr/lib/python3.8
  /usr/lib/python3.8/lib-dynload
  /home/USER/.local/lib/python3.8/site-packages
  /usr/local/lib/python3.8/dist-packages
  /usr/lib/python3/dist-packages
[Warn  - 4:11:49 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Warn  - 4:11:49 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Warn  - 4:11:49 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Warn  - 4:11:49 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Warn  - 4:11:49 PM] Exception received when installing file system watcher: TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
[Info  - 4:11:49 PM] Searching for source files
[Info  - 4:11:49 PM] Found 130 source files
[FG] parsing: /home/USER/repos/REPO/test_pylance.py (16ms)
[FG] parsing: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (98ms)
[FG] binding: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/builtins.pyi (41ms)
[FG] binding: /home/USER/repos/REPO/test_pylance.py (1ms)
[Info  - 4:11:49 PM] Background analysis(1) root directory: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist
[Info  - 4:11:49 PM] Background analysis(1) started
Background analysis message: setConfigOptions
Background analysis message: setImportResolver
Background analysis message: ensurePartialStubPackages
Background analysis message: setTrackedFiles
Background analysis message: markAllFilesDirty
Background analysis message: setFileOpened
Background analysis message: analyze
[BG(1)] analyzing: /home/USER/repos/REPO/test_pylance.py ...
[BG(1)]   parsing: /home/USER/repos/REPO/test_pylance.py (28ms)
[BG(1)]   parsing: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 4ms] (103ms)
[BG(1)]   binding: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/builtins.pyi (39ms)
[BG(1)]   binding: /home/USER/repos/REPO/test_pylance.py (0ms)
[BG(1)]   checking: /home/USER/repos/REPO/test_pylance.py ...
[BG(1)]     parsing: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/typing.pyi [fs read 2ms] (31ms)
[BG(1)]     binding: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/typing.pyi (11ms)
[BG(1)]     parsing: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (8ms)
[BG(1)]     binding: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (1ms)
[BG(1)]     parsing: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 0ms] (2ms)
[BG(1)]     binding: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/typing_extensions.pyi (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/hist/__init__.py [fs read 0ms] (4ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/hist/__init__.py (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/hist/hist.py [fs read 1ms] (2ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/hist/hist.py (0ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/hist/basehist.py [fs read 1ms] (44ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/hist/basehist.py (9ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/__init__.py [fs read 0ms] (5ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/__init__.py (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/hist.py [fs read 0ms] (50ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/hist.py (12ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/utils.py [fs read 0ms] (9ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/utils.py (2ms)
[BG(1)]     parsing: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/logging/__init__.pyi [fs read 0ms] (26ms)
[BG(1)]     binding: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/typeshed-fallback/stdlib/logging/__init__.pyi (8ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/typing.py [fs read 0ms] (8ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/typing.py (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_core/hist.pyi [fs read 0ms] (4ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_core/hist.pyi (2ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_core/__init__.pyi [fs read 1ms] (1ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_core/__init__.pyi (0ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/hist/quick_construct.py [fs read 0ms] (4ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/hist/quick_construct.py (2ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/hist/axis/__init__.py [fs read 0ms] (9ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/hist/axis/__init__.py (4ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/hist/storage.py [fs read 0ms] (2ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/hist/storage.py (0ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/storage.py [fs read 0ms] (0ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/storage.py (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/storage.py [fs read 0ms] (0ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/storage.py (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/__init__.py [fs read 1ms] (16ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/numpy/__init__.py ...
[BG(1)]       parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/__init__.py [fs read 0ms] (26ms)
[BG(1)]       binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/__init__.py ...
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/numeric.py [fs read 0ms] (46ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/numeric.py ...
[BG(1)]           parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/umath.py [fs read 0ms] (8ms)
[BG(1)]           binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/umath.py ...
[BG(1)]             parsing: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/bundled/native-stubs/numpy/core/_multiarray_umath.pyi [fs read 1ms] (59ms)
[BG(1)]             binding: /home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/bundled/native-stubs/numpy/core/_multiarray_umath.pyi (22ms)
[BG(1)]           binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/umath.py (82ms)
[BG(1)]           parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/numerictypes.py [fs read 0ms] (14ms)
[BG(1)]           binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/numerictypes.py (3ms)
[BG(1)]           parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/fromnumeric.py [fs read 0ms] (20ms)
[BG(1)]           binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/fromnumeric.py (7ms)
[BG(1)]           parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/arrayprint.py [fs read 0ms] (17ms)
[BG(1)]           binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/arrayprint.py (11ms)
[BG(1)]           parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/_asarray.py [fs read 0ms] (2ms)
[BG(1)]           binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/_asarray.py (1ms)
[BG(1)]           parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/_ufunc_config.py [fs read 0ms] (3ms)
[BG(1)]           binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/_ufunc_config.py (2ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/numeric.py (183ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/records.py [fs read 0ms] (13ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/records.py (6ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/memmap.py [fs read 0ms] (5ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/memmap.py (2ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/function_base.py [fs read 0ms] (10ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/function_base.py (2ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/machar.py [fs read 0ms] (6ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/machar.py (4ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/getlimits.py [fs read 0ms] (10ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/getlimits.py (3ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/shape_base.py [fs read 1ms] (13ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/shape_base.py (3ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/einsumfunc.py [fs read 1ms] (13ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/einsumfunc.py (8ms)
[BG(1)]       binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/__init__.py (331ms)
[BG(1)]       parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/__init__.py [fs read 0ms] (2ms)
[BG(1)]       binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/__init__.py ...
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/type_check.py [fs read 0ms] (4ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/type_check.py (1ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/index_tricks.py [fs read 0ms] (9ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/index_tricks.py (5ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/function_base.py [fs read 0ms] (41ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/function_base.py (16ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/nanfunctions.py [fs read 0ms] (13ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/nanfunctions.py (3ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/shape_base.py [fs read 1ms] (14ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/shape_base.py (3ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/stride_tricks.py [fs read 1ms] (3ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/stride_tricks.py (1ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/twodim_base.py [fs read 0ms] (5ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/twodim_base.py (1ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/ufunclike.py [fs read 0ms] (1ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/ufunclike.py (0ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/histograms.py [fs read 0ms] (7ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/histograms.py (3ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/polynomial.py [fs read 1ms] (12ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/polynomial.py (8ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/utils.py [fs read 0ms] (13ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/utils.py (6ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/arraysetops.py [fs read 0ms] (7ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/arraysetops.py (3ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/npyio.py [fs read 1ms] (28ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/npyio.py (14ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/financial.py [fs read 0ms] (17ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/financial.py (3ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/arraypad.py [fs read 1ms] (5ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/arraypad.py (2ms)
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/_version.py [fs read 1ms] (2ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/_version.py (0ms)
[BG(1)]       binding: /home/USER/.local/lib/python3.8/site-packages/numpy/lib/__init__.py (255ms)
[BG(1)]       parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/matrixlib/__init__.py [fs read 0ms] (1ms)
[BG(1)]       binding: /home/USER/.local/lib/python3.8/site-packages/numpy/matrixlib/__init__.py ...
[BG(1)]         parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/matrixlib/defmatrix.py [fs read 0ms] (3ms)
[BG(1)]         binding: /home/USER/.local/lib/python3.8/site-packages/numpy/matrixlib/defmatrix.py (2ms)
[BG(1)]       binding: /home/USER/.local/lib/python3.8/site-packages/numpy/matrixlib/__init__.py (5ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/numpy/__init__.py (625ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/numpy/core/multiarray.py [fs read 0ms] (8ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/numpy/core/multiarray.py (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/axis/__init__.py [fs read 0ms] (0ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/axis/__init__.py (1ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/axis.py [fs read 0ms] (6ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_internal/axis.py (6ms)
[BG(1)]     parsing: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_core/axis/__init__.pyi [fs read 0ms] (1ms)
[BG(1)]     binding: /home/USER/.local/lib/python3.8/site-packages/boost_histogram/_core/axis/__init__.pyi (1ms)
[Error - 4:11:51 PM] An internal error occurred while type checking file "/home/USER/repos/REPO/test_pylance.py": RangeError: Maximum call stack size exceeded
    at partiallySpecializeFunctionForBoundClassOrObject (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:24206:14)
    at bindFunctionToClassOrObject (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:24107:24)
    at forEach (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20800:45)
    at Array.forEach (<anonymous>)
    at canAssignProperty (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20786:19)
    at forEach (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20540:34)
    at Map.forEach (<anonymous>)
    at canAssignClassToProtocol (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20446:25)
    at canAssignClass (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20992:20)
    at Ai (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:22475:22)
[BG(1)]   checking: /home/USER/repos/REPO/test_pylance.py (1389ms)
[BG(1)] analyzing: /home/USER/repos/REPO/test_pylance.py (1563ms)
[Error - 4:11:51 PM] Error performing analysis: RangeError: Maximum call stack size exceeded
    at partiallySpecializeFunctionForBoundClassOrObject (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:24206:14)
    at bindFunctionToClassOrObject (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:24107:24)
    at forEach (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20800:45)
    at Array.forEach (<anonymous>)
    at canAssignProperty (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20786:19)
    at forEach (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20540:34)
    at Map.forEach (<anonymous>)
    at canAssignClassToProtocol (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20446:25)
    at canAssignClass (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20992:20)
    at Ai (/home/USER/.vscode/extensions/ms-python.vscode-pylance-2021.8.1/dist/pyright-internal/src/analyzer/typeEvaluator.ts:22475:22)

```
</details>

## Code Snippet / Additional information

A minimal example would be (wich is also the content of `/home/USER/repos/REPO/test_pylance.py`):
```python
import hist

hist.Hist(hist.axis.Integer(0, 10))
```

With [`hist`](https://github.com/scikit-hep/hist) being installed via `pip` in it's current version `2.4.0` with the below listed dependency versions:
```
boost-histogram==1.1.0
numpy==1.19.1
histoprint==2.1.0
click==7.1.2
```

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem, and I'll take a look at it.

**Comment by erictraut** [comment]
I found the cause, and the fix will be included in the next release.

**Comment by heejaechang** [comment]
it should be fixed in 2021.8.2 released today (2021.8.19)
